app/src/main/java/ch/epfl/qedit/model/MatrixFormat.java

```diff
@@ -0,0 +1,74 @@
+package ch.epfl.qedit.model;
+
+public class MatrixFormat extends AnswerFormat {
+    private Visitor visitor;
+    private boolean hasDecimal = true;
+    private boolean hasSign = true;
+
+    private int tableRowsNumber = 1;
+    private int tableColumnsNumber = 1;
+    private int maxCharacters = 3;
+    private String hintString = hint();
+
+    public MatrixFormat(
+            int tableColumnsNumber,
+            int tableRowsNumber,
+            boolean hasDecimal,
+            boolean hasSign,
+            int maxCharacters) {
+        super();
+        this.tableColumnsNumber = tableColumnsNumber;
+        this.tableRowsNumber = tableRowsNumber;
+        this.hasDecimal = hasDecimal;
+        this.hasSign = hasSign;
+        this.maxCharacters = maxCharacters;
+    }
+
+    public static MatrixFormat createMatrix33(
+            boolean hasDecimal, boolean hasSign, int maxCharacters) {
+        return new MatrixFormat(3, 3, hasDecimal, hasSign, maxCharacters);
+    }
+
+    public static MatrixFormat createSingleField(
+            boolean hasDecimal, boolean hasSign, int maxCharacters) {
+        return new MatrixFormat(1, 1, hasDecimal, hasSign, maxCharacters);
+    }
+
+    @Override
+    public void accept(Visitor visitor) {
+        this.visitor = visitor;
+    }
+
+    public int getTableRowsNumber() {
+        return tableRowsNumber;
+    }
+
+    public int getTableColumnsNumberNumber() {
+        return tableColumnsNumber;
+    }
+
+    public boolean getHasDecimal() {
```

Comment:
Sure, but I don't see how it would be clearer.

Reply:
I know I'm picky ;), but chaining two verbs sounds a bit weird IMO. Get and Has kind of clash together. That's a minor issue, I just noted it because I care about details.

Reply:
Yeah, it is fine with. I've changed it